### PR TITLE
Ensure dominant speaker streams are rendered when under OVC limit

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What
Ensure dominant speaker streams are rendered when OVC is less than the total remote streams

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3766785

# How Tested
Added 2 failing unit tests which passed after the changes in this PR

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/VideoGallery.test.tsx [hunk 1]
```diff
@@ -678,6 +678,92 @@ describe('VideoGallery with vertical overflow gallery tests', () => {
...
+    expect(verticalGalleryTiles.length).toBe(4);
+    expect(getDisplayName(verticalGalleryTiles[0])).toBe('Remote Participant 0');
+    expect(tileIsVideo(verticalGalleryTiles[0])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[1])).toBe('Remote Participant 1');
+    expect(tileIsVideo(verticalGalleryTiles[1])).toBe(false);
+    expect(getDisplayName(verticalGalleryTiles[2])).toBe('Remote Participant 2');
+    expect(tileIsVideo(verticalGalleryTiles[2])).toBe(true);
+    expect(getDisplayName(verticalGalleryTiles[3])).toBe('Remote Participant 3');
+    expect(tileIsVideo(verticalGalleryTiles[3])).toBe(true);
+  });
```
Comment (mgamis-msft) [inline packages/react-components/src/components/VideoGallery.test.tsx:766]: Add test for spotlighted participant is showing their stream and also a spotlighted participant without their stream showing

Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery.test.tsx:766]: Also for the other layouts

Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery.test.tsx:766]: ... in another PR

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-components/src/components/VideoGallery.test.tsx [hunk 2]
```diff
@@ -678,6 +678,92 @@ describe('VideoGallery with vertical overflow gallery tests', () => {
...
+    const remoteParticipants = [...Array(10).keys()].map((i) => {
+      return createRemoteParticipant({
+        userId: `${i}`,
+        displayName: `Remote Participant ${i}`,
+        videoStream: i < 5 ? { isAvailable: true, renderElement: createVideoDivElement() } : undefined
+      });
+    });
+
+    const videoGalleryProps: VideoGalleryProps = {
+      layout: 'floatingLocalVideo',
```
[the remark this answers is not in the document]
Comment (mgamis-msft) [reply packages/react-components/src/components/VideoGallery.test.tsx:743]: Definitely

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**